cpp/oneapi/dal/algo/knn/backend/gpu/infer_kernel_brute_force_dpc.cpp

```diff
@@ -0,0 +1,96 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#define DAAL_SYCL_INTERFACE
+#include "src/algorithms/k_nearest_neighbors/oneapi/bf_knn_classification_predict_kernel_ucapi.h"
+#include "data_management/data/numeric_table.h"
+
+#include "oneapi/dal/algo/knn/backend/gpu/infer_kernel.hpp"
+#include "oneapi/dal/algo/knn/backend/model_interop.hpp"
+#include "oneapi/dal/algo/knn/detail/model_impl.hpp"
+#include "oneapi/dal/backend/interop/common.hpp"
+#include "oneapi/dal/backend/interop/table_conversion.hpp"
+#include "oneapi/dal/detail/common.hpp"
+
+namespace oneapi::dal::knn::backend {
+
+using dal::backend::context_gpu;
+using namespace daal::data_management;
+
+namespace daal_knn = daal::algorithms::bf_knn_classification;
+namespace interop  = dal::backend::interop;
+
+template <typename Float>
+using daal_knn_brute_force_kernel_t = daal_knn::prediction::internal::
+    KNNClassificationPredictKernelUCAPI<Float>;
+
+template <typename Float>
+static infer_result call_daal_kernel(const context_gpu& ctx,
+                                     const descriptor_base& desc,
+                                     const table& data,
+                                     model m) {
+    const std::int64_t row_count    = data.get_row_count();
+    const std::int64_t column_count = data.get_column_count();
+
+    auto arr_data  = row_accessor<const Float>{ data }.pull();
+    auto arr_labels = array<Float>::empty(1 * row_count);
+    // TODO: read-only access performed with deep copy of data since daal numeric tables are mutable.
+    // Need to create special immutable homogen table on daal interop side
+
+    // TODO: data is table, not a homogen_table. Think better about accessor - is it enough to have just a row_accessor?
+    const auto daal_data =
+        interop::convert_to_daal_homogen_table(arr_data, row_count, column_count);
+    const auto daal_labels = interop::convert_to_daal_homogen_table(arr_labels, row_count, 1);
+
+    const std::int64_t dummy_seed = 777;
```

Comment:
Where it is used? Seems to be futile.

Reply:
It will be replaced with a status. There is a TODO comment below